### PR TITLE
fix lookup bug

### PR DESCRIPTION
**Pull request by CorgiMan**
@uber/ringpop

### CODE REVIEW

File: hashring/hashring_test.go [hunk 1]
```diff
@@ -177,6 +177,80 @@ func TestLookup(t *testing.T) {
...
+	}
+
+	servers := make(map[string]bool)
+	for _, key := range keys {
+		server, ok := ring.Lookup(key)
+		assert.True(t, ok, "expected that the lookup is a success")
+		servers[server] = true
+	}
+
+	// Expect no more than one colission
```
Comment (danielheller) [inline hashring/hashring_test.go]: "Collision"

File: hashring/hashring_test.go [hunk 2]
```diff
@@ -177,6 +177,80 @@ func TestLookup(t *testing.T) {
...
+	assert.Len(t, servers, len(keys), "expected that all keys are owned by a different server")
+}
+
+// TestLookupNNoGaps tests the selected servers from lookup N are
```
Comment (danielheller) [inline hashring/hashring_test.go]: "tests the selected servers from lookup N are form" => "tests that the servers selected by lookupN form"

File: hashring/hashring_test.go (hunk 1 above)
Comment (danielheller) [inline hashring/hashring_test.go]: Comment disagrees with assertion.

Comment (CorgiMan) [inline hashring/hashring_test.go]: ack

File: hashring/hashring_test.go (hunk 2 above)
Comment (CorgiMan) [inline hashring/hashring_test.go]: ack

File: hashring/hashring_test.go [hunk 3]
```diff
@@ -177,6 +177,86 @@ func TestLookup(t *testing.T) {
...
+	// or larger than the values of the servers that are part of the lookup.
+	allExcluded := true
+	for _, ad := range addresses {
+		if _, ok := serversSet[ad]; ok {
+			continue
+		}
+		adHash := ring.hashfunc(ad + "0")
+		if adHash >= min && adHash <= max {
+			allExcluded = false
+			fmt.Println(adHash, min, max)
```
Comment (jwolski) [inline hashring/hashring_test.go]: get rid of the println plz

File: hashring/hashring_test.go [hunk 4]
```diff
@@ -177,6 +177,86 @@ func TestLookup(t *testing.T) {
...
+	// We are reconstructing the values under which the servers are stored in
+	// the red-black tree. This approach is brittle but it gives us deeper
+	// introspection into the internals of the tree. The hashring is configured
+	// to only store one replica per server, if we didn't, it would be
+	// impossible to find out which specific replica has been iterated over
+	// by LookupN.
+	hashes := make([]int, 0, len(servers))
+	for _, s := range servers {
+		// We are appending 0 to get the first (and only) replica of the server.
+		hashes = append(hashes, ring.hashfunc(s+"0"))
```
Comment (jwolski) [inline hashring/hashring_test.go]: instead of using hashfunc why not use the rings public Lookup API?

**Comment by jwolski** [comment]
lgtm

**Pull request merged**